scapy/sendrecv.py

```diff
@@ -872,15 +871,22 @@ def sniff(count=0, store=True, offline=None, prn=None, lfilter=None,
     remain = None
 
     # Get select information from the sockets
-    _main_socket = next(iter(sniff_sockets))
-    read_allowed_exceptions = _main_socket.read_allowed_exceptions
-    select_func = _main_socket.select
-    # We check that all sockets use the same select(), or raise a warning
-    if not all(select_func == sock.select for sock in sniff_sockets):
-        warning("Warning: inconsistent socket types ! The used select function"
-                "will be the one of the first socket")
-    # Now let's build the select function, used later on
-    _select = lambda sockets, remain: select_func(sockets, remain)[0]
+    sniff_sockets_t = list(set([type(x) for x in sniff_sockets.keys()]))
+    group_sockets = lambda sockets: [[x for x in sockets.keys()
+                                      if type(x) == sniff_sockets_t[i]]
+                                     for i in range(len(sniff_sockets_t))]
```

Comment:
You can use enumerate

Reply:
enumerate will cause this error: 
```
Traceback (most recent call last):
  File "<input>", line 13, in <module>
  File "/home/nils/PycharmProjects/secdev_scapy/scapy/sendrecv.py", line 878, in sniff
    select_funcs = [x[0].select for x in group_sockets(sniff_sockets)]
  File "/home/nils/PycharmProjects/secdev_scapy/scapy/sendrecv.py", line 877, in <lambda>
    for i in enumerate(sniff_sockets_t)]
  File "/home/nils/PycharmProjects/secdev_scapy/scapy/sendrecv.py", line 877, in <listcomp>
    for i in enumerate(sniff_sockets_t)]
  File "/home/nils/PycharmProjects/secdev_scapy/scapy/sendrecv.py", line 876, in <listcomp>
    if isinstance(sniff_sockets_t[i], x)]
TypeError: list indices must be integers or slices, not tuple
```

Reply:
I could replace `range(len(` here, but I don't see a way to improve line 880

Reply:
Here’s a syntax example
```
for key, value in enumerate(some_list):
    print(counter, value)
```
It provides both index and key at the same time